vllm/model_executor/layers/rotary_embedding.py

```diff
@@ -507,6 +617,21 @@ def get_rope(
                                                     base, is_neox_style,
                                                     scaling_factor,
                                                     **extra_kwargs)
+        elif scaling_type == "deepseek_yarn":
+            original_max_position = rope_scaling[
+                "original_max_position_embeddings"]
+            # assert max_position == original_max_position * scaling_factor
+            extra_kwargs = {
+                k: v
+                for k, v in rope_scaling.items()
+                if k in ("extrapolation_factor", "attn_factor", "beta_fast",
+                         "beta_slow", "mscale", "mscale_all_dim")
+            }
+            rotary_emb = DeepseekScalingRotaryEmbedding(head_size, rotary_dim,
+                                                    original_max_position,
```

Comment:
It can be confusing to call the `get_rope` with config.max_position_embedding, while generating the rope cache with config.rope_scaling['original_max_position_embeddings']. It's better to directly pass the right `max_position`.